crates/block-producer/src/runner.rs

```diff
@@ -338,3 +348,61 @@ pub fn run(config: Config) -> Result<()> {
 
     Ok(())
 }
+
+fn check_ckb_version(rpc_client: &RPCClient) -> Result<()> {
+    let ckb_version = block_on(rpc_client.get_ckb_version())?;
```

Comment:
```suggestion
    let ckb_version = smol::block_on(rpc_client.get_ckb_version())?;
```

Pls also fix other block_on calling in this PR.